website/docs/dependency-pinning.md

```diff
@@ -103,12 +103,14 @@ Lock files are a great companion to semver ranges _or_ pinning dependencies, bec
 
 #### What a lock file will do for you
 
-A lock file will lock down the exact dependencies and _sub_-dependencies that your project uses, so that everyone running `npm install` or `yarn install` will install the exact same dependencies as the person or bot that last updated the lock file.
+When kept in sync with its associated `package.json`, a lock file will further lock down the exact dependencies and _sub_-dependencies that your project uses, so that everyone running `npm install` or `yarn install` will install the exact same dependencies as the person or bot that last updated the lock file.
```

Comment:
I'm not completely sure what "kept in sync" means here, and I don't think it fully explains the behaviour.

One common situation is: (1) user A run `npm install`, `package-lock.json` is updated (in this case, I would consider `package.json` and `package-lock.json` to be in sync); (2) after pulling the repo, user B runs `npm install`, which then updates `package-lock.json`.

This behaviour is implied by the [blog post announcing `npm ci`](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable): "Previously, developers who wanted to ensure that node_modules/ and package.json stayed in sync would have to archive their node_modules folder." (That is, prior to the `npm ci` command, _there was no way_ to ensure `node_modules/` and `package.json` stayed in sync, even with `package-lock.json`.)

Reply:
The command `npm install` should always "sync" the `package-lock.json` to the `package.json` if they weren't already in sync from the previous install. Note that I'm deliberately saying `npm install` and not something like `npm install react`.

Both your scenarios (lock file being updated during a regular `npm install`) should only happen if the `package.json` was modified - and the `package-lock.json` not modified - after the previous `npm install`. Otherwise `package-lock.json` should not be modified by any `npm install` (it's a bug in `npm` CLI if so, not a feature).

Under normal scenarios, `npm install` and `npm ci` should produce the same `node_modules`. In abnormal scenarios where the lock file is out of sync, the difference is:
- `npm install` will proceed, and update the `package-lock.json` to match
- `npm ci` will abort and throw an error

The second scenario is therefore recommended for developers and CI, so that you are alerted immediately of any lock file sync problem. The equivalent in `yarn` for `npm ci` is `yarn install --frozen-lockfile`.